File: Códigos/exercicios_aula_01.R
# Exemplo 1: Lancamento de dado

exemplo1 <- function(){
  nsamples = 12
  
  dado = c(1:6)
  nFaces = length(dado)
  
  totalFaces = rep(0, nFaces)
  
  for (i in 1:nsamples){
    face = sample(dado, 1)
    totalFaces[face] = totalFaces[face] + 1
  }
  
  print("Exemplo 1: Lançamento de dado")
  for (i in 1:nFaces){
    print(paste("Total da face [", i, "] = ", totalFaces[i]))
  }
  
  for (i in 1:nFaces){
    print(paste("Probabilidade da face [", i, "] = ", totalFaces[i]/nsamples))
  }
}

# Exemplo 2: Lancamento de moeda
exemplo2 <- function(){
  nsamples = 10000
  
  moeda = c('ca', 'co')
  nFaces = length(moeda)
  
  cont.evento.a = 0
  cont.evento.b = 0
  
  for (i in 1:nsamples){
    face1 = sample(moeda, 1, prob=c(2/5, 3/5))
    face2 = sample(moeda, 1, prob=c(2/5, 3/5))
    face3 = sample(moeda, 1, prob=c(2/5, 3/5))
    
    # Evento A
    if (((face1 == 'co') & (face2 == 'ca')) | ((face1 == 'ca') & (face2 == 'co')))
      cont.evento.a = cont.evento.a + 1
    
    # Evento B
    if (face2 == 'co' & face3 == 'ca')
      cont.evento.b = cont.evento.b + 1
  }
  
  PA = cont.evento.a / nsamples
  PB = cont.evento.b / nsamples
  
  print(PA)
  print(PB)
}

exemplo2()

# Exemplo 3: Adubo

File: Códigos/exercicios_aula01.py
# -*- coding: utf-8 -*-
"""
Created on Thu Mar  7 09:26:52 2024

@author: User
"""
import random, numpy

def exemplo1(nsamples):
    dado = [i+1 for i in range(6)]
    nFaces = len(dado)
    
    totalFaces = [0 for _ in range(6)]
    
    for i in range(nsamples):
        face = random.choice(dado)
        totalFaces[face-1] += 1
    
    print("-----> Exemplo 1: Lancamento de dado\n")
    for i in range(nFaces):
        print("Total da face [{}] = {}".format(i+1, totalFaces[i]))
        print("Probabilidade da face [{}] = {}\n".format(i+1, totalFaces[i]/nsamples))
    return

def exemplo2(nsamples):
    moeda = ['CA', 'CO']
    prob = [2/5, 3/5]
    
    contA = 0
    contB = 0
    
    for _ in range(nsamples):
        # random.choices retorna uma lista de tamanho k
        face1 = random.choices(moeda, weights=prob, k=1)[0]
        face2 = random.choices(moeda, weights=prob, k=1)[0]
        face3 = random.choices(moeda, weights=prob, k=1)[0]
        
        if ((face1 == 'CA' and face2 == 'CO') or (face1 == 'CO' and face2 == 'CA')): contA += 1
            
        if (face2 == 'CO' and face3 == 'CO'): contB += 1
            
    PA = contA / nsamples
    PB = contB / nsamples
    
    print(PA)
    print(PB)
    return

def exemplo3(nsamples):
    return


def main():
    # exemplo1(12)
    # exemplo2(10000)
    return

main()